moveos/moveos-verifier/src/verifier.rs

```diff
@@ -40,76 +42,107 @@ where
                         fdef,
                         module.self_id(),
                     ));
                 }
-                for ref ty in loaded_function.parameters {
-                    match ty {
-                        Type::Reference(bt) | Type::MutableReference(bt) => match bt.as_ref() {
-                            Type::Struct(s) | Type::StructInstantiation(s, _) => {
-                                let struct_type = session.get_struct_type(*s).unwrap();
-                                if !check_storage_context_struct_tag(format!(
-                                    "{}::{}::{}",
-                                    struct_type.module.short_str_lossless(),
-                                    struct_type.module.name(),
-                                    struct_type.name
-                                )) {
-                                    return Err(vm_error_for_init_func_checking(
-                                            StatusCode::TYPE_MISMATCH,
-                                            "init function should not input structures other than storageContext",
-                                            fdef,
-                                            module.self_id(),
-                                        ));
-                                }
-                            }
-                            Type::Signer => {}
-                            _ => {
-                                return Err(vm_error_for_init_func_checking(
-                                        StatusCode::TYPE_MISMATCH,
-                                        "init function should only enter reference signer or mutable reference storageContext",
-                                        fdef,
-                                        module.self_id(),
-                                    ));
-                            }
-                        },
-                        Type::Signer => {}
-                        _ => {
-                            return Err(vm_error_for_init_func_checking(
-                                StatusCode::TYPE_MISMATCH,
-                                "init function should only enter signer or storageContext",
-                                fdef,
-                                module.self_id(),
-                            ))
-                        }
+
+                if let Some(struct_full_name) = struct_name_opt {
+                    if !check_storage_context_struct_tag(struct_full_name) {
+                        return Err(vm_error_for_init_func_checking(
+                            StatusCode::TYPE_MISMATCH,
+                            "init function should not input structures other than storageContext",
+                            fdef,
+                            module.self_id(),
+                        ));
                     }
                 }
             }
+
+            return Ok(true);
```

Comment:
If the module does not contain the `init` function, is it also returns `Ok(true)`?